apps/api/src/app/events/usecases/add-job/should-add-digest-job.usecase.ts

```diff
@@ -14,14 +15,19 @@ export class ShouldAddDigestJob {
     if (!isValidDigestStep || !data) {
       return true;
     }
-
-    const delayedDigest = await this.jobRepository.findOne({
+    const digestKey = data?.step?.metadata?.digestKey;
+    const where: any = {
       status: JobStatusEnum.DELAYED,
       type: StepTypeEnum.DIGEST,
       _subscriberId: data._subscriberId,
       _templateId: data._templateId,
       _environmentId: data._environmentId,
-    });
+    };
+    if (digestKey) {
+      //where['payload.' + digestKey] = data.payload[digestKey];
+      where['payload.' + digestKey] = DigestFilterSteps.getNestedValue(data.payload, digestKey);
+    }
+    const delayedDigest = await this.jobRepository.findOne(where);
```

Comment:
In the PR where I am fixing the digest job problems reported we are moving this to the DAL. I will take care of it as this use case will disappear.